### ISSUE TITLE
MariaDB truncation warning is treated as an error

### ISSUE THREAD

**Issue by cookieguru** [HeidiSQL/HeidiSQL#2007]
### What where you trying to do?
Edit a cell's value in the Data grid.  See reproduction steps below

### What actually happened?
MariaDB updated the value and issued a warning, but HeidiSQL treated it as an error

### What did you expect to happen?
The warning would be presented, the row is no longer marked unsaved, and the actual value is shown in the appropriate cell

### What do you think actually happened?
The MariaDB driver reported that 0 rows changed when in fact 1 row changed
https://github.com/HeidiSQL/HeidiSQL/blob/f32abebeff177552e4b594c3d89d8fb6369da650/source/dbconnection.pas#L9773-L9774

### How do other DBMSes handle this?
Performing the same steps on MySQL 8.0.39 an error is thrown: `/* SQL Error (1406): Data too long for column 'foo' at row 1 */` and the row is not updated, and the value is not truncated.  No other DBMSes have been tested

### HeidiSQL version

12.8.0.6919

### Database server version

10.3.39-MariaDB-cll-lve

### Reproduction recipe

Assume the following table and data:
```sql
CREATE TABLE `foo` (
  `bar` varchar(3) NOT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;

INSERT INTO `foo` (`bar`) VALUES ('abc');
```

1. Open the table Data view and change the row's value to `1234`.
2. Blur the row to cause the edit to be committed
3. See the error message `0 rows updated when that should have been 1.`
4. Note the following in the query log:
```sql
UPDATE `database`.`foo` SET `bar`='1234' WHERE  `bar`='abc' LIMIT 1;
SHOW WARNINGS;
/* Warning: (1265) Data truncated for column 'bar' at row 1 */
```
5. Hit the escape key to remove the triangle from the cell.  Note the cell shows `abc`
6. Run `SELECT * FROM foo` and see the value `123` and not `abc`

**Comment by ansgarbecker** [comment]
In order to turn the error into a warning, you need the server variable `sql_mode` to exclude `STRICT_TRANS_TABLES`. Before I removed that, I got an error, and I guess that's the same in newer versions, including MySQL.

When trying to reproduce now, I see the value is not truncated, and after refreshing it has the previous value:

![grid-data-too-long](https://github.com/user-attachments/assets/6ccdfad3-4f8d-4e99-a593-f29ca2a77ac8)

So the point 6 in your reproduction recipe is different here.

Could this be caused by something different, probably master/slave synchronisation?
Can you please post the result of `SELECT @@SESSION.sql_mode;` ?

**Comment by cookieguru** [comment]
> Can you please post the result of `SELECT @@SESSION.sql_mode;` ?

`NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION`

**Comment by ansgarbecker** [comment]
I just found the cause of it, and saw I already left a "TODO" line in the code, where I forgot to test. That was broken since I solved issue #1965.
Next build fixes the wrong (0) rows affected to be used, which removes the error dialog mentioned above. Also, the changed value (e.g. "1234") is then displayed in the grid, which is still not really correct, but at least that's how it was before.
